pyccel/codegen/printing/ccode.py

```diff
@@ -268,6 +368,87 @@ def function_signature(self, expr):
             arg_code = ', '.join('{0}{1}'.format(self.get_declare_type(i), i) for i in expr.arguments)
         return '{0}{1}({2})'.format(ret_type, name, arg_code)
 
+    def _print_NumpyUfuncBase(self, expr):
+        """ convert a python expresion with a numpy function call to c
+        function call
+
+        Parameters:
+        -----------
+            expr (Pyccel ast node): python expresion with a numpy function call
+
+        Returns:
+        --------
+            string: equivalent of the expression in c language
+
+        Example:
+        -------
+            numpy.cos(x) ==> cos(x)
+
+        """
+        # add necessary include
+        self._additional_imports.add('math.h')
+        type_name = type(expr).__name__
+        func_name = numpy_ufunc_to_c[type_name]
+        args = []
+        for arg in expr.args:
+            if arg.dtype is not NativeReal():
+                args.append('(' + self._print_PythonFloat(arg) + ')')
```

Comment:
The print functions are designed to be passed an instance of the class indicated in the name. So `_print_PythonFloat` should only ever be passed a `PythonFloat` as an argument. `PythonFloat` is the class that takes care of the cast while `_print_PythonFloat` takes care of printing a `PythonFloat`. So this call should either be:
`self._print_PythonFloat(PythonFloat(arg))`
Or as in the rest of the code:
`self._print(PythonFloat(arg))`

The second option is usually preferred as it allows the `_print` function to select the right function call. This means that the code will still work if for example one day we decided to implement the printing of all casts via a superclass `CastFunc` then when `_print_PythonFloat` was removed `_print` would know to call `_print_CastFunc` instead but anything calling `_print_PythonFloat` directly would just stop working

Does that make sense?